### PR TITLE
File upload support for Flask

### PR DESCRIPTION
**Pull request by schiism**
## Description

Add support for file uploads in multipart forms for Flask

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #327 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: tests/flask/app.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import strawberry
+from flask import Flask
+from strawberry.file_uploads import Upload
+from strawberry.flask.views import GraphQLView as BaseGraphQLView
+
+
+def create_app(**kwargs):
```
Comment (patrick91) [inline tests/flask/app.py:7]: can you make this a fixture?

File: tests/flask/test_upload.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        },
+        content_type="multipart/form-data",
+    )
+
+    assert response.status_code == 200
+
+    data = json.loads(response.data.decode())
+
+    assert not data.get("errors")
+    assert data["data"]["readText"] == "strawberry"
```
Comment (patrick91) [inline tests/flask/test_upload.py:39]: 👍

File: tests/flask/test_upload.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+from .app import create_app
+
+
+@pytest.fixture
+def flask_client():
+    app = create_app()
+
+    with app.test_client() as client:
+        yield client
```
Comment (patrick91) [inline tests/flask/test_upload.py:14]: I think this is duplicated now :) you can move this fixture inside `tests/flask/conftest.py` to make it available in the flask tests

**Review by patrick91 — COMMENTED** [review]
Nice work! I left a couple of minor comments, after that I think we can merge :)

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
```
Comment (BryceBeagle) [inline RELEASE.md:1]: As this is a new feature, this should probably be a minor version bump?

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: tests/flask/app.py (hunk 1 above)
Comment (schiism) [reply tests/flask/app.py:7]: I'm going to leave this one as it is since it gets called with a kwarg

Comment (patrick91) [reply tests/flask/app.py:7]: yes, I didn't notices it is used in multiple places 😊

File: docs/features/file-upload.md [hunk 5]
```diff
@@ -26,6 +26,23 @@ class Mutation:
...
 ```
 
+## Flask
+
+Example:
+
+```python
+import strawberry
+from strawberry.file_uploads import Upload
+...
```
Comment (patrick91) [inline docs/features/file-upload.md:36]:
```suggestion


```

**Review by patrick91 — APPROVED** [review]
🔥

**Pull request merged**